pfcpiface/conn.go

```diff
@@ -148,8 +148,20 @@ func (pConn *PFCPConn) Serve() {
 
 				continue
 			}
+			buf = buf[:n]
+
+			// Check that the PFCP message did not get truncated.
+			header, err := message.ParseHeader(buf)
+			if err != nil {
+				log.Errorln("could not parse PFCP message header: ", err)
+				continue
+			}
+			if header.Length > uint16(n) {
+				log.Errorln("receive buffer too small for PFCP message")
+				continue
+			}
```

Comment:
This check is already happening in the library: https://github.com/omec-project/upf-epc/blob/a6b6a4e29cc56c902cf7ca03b10a7d01f9cd98ae/pfcpiface/vendor/github.com/wmnsk/go-pfcp/message/session-establishment-request.go#L281-L286